toolchain/README.md

```diff
@@ -0,0 +1,168 @@
+# Toolchain
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+The toolchain represents the production portion of Carbon. The toolchain's top
+priority is performance: it needs to generate code quickly and scale well.
```

Comment:
I think "top priority" is too strong here. Could we instead describe performance as a high priority, or perhaps as a major (the most significant?) difference between the toolchain and explorer? Or perhaps we can summarize all the major differences between toolchain and explorer here (performance, code generation, C++ interop)?

Brainstorming goals for the toolchain and prioritizing them, I get:

1. Correctness.
2. Quality of generated code.
3. Performance.
4. Quality of diagnostics for incorrect or questionable code.

... and I think I might want to rank them in different orders for different build modes (for the development build mode I might put performance at the bottom of the above list).

Reply:
Added priorities with some small edits, avoiding putting on numbers so there's less of a feeling of strong ranking (particularly for your opt vs dbg point)